### PR TITLE
Add note for installing neuroconv datainterfaces with `zsh`

### PR DESCRIPTION
**Pull request by weiglszonja**
## Motivation

I'm using `zsh` and noticed that installing neuroconv as `pip install neuroconv[extract]` did not work for me. The solution was to use quotes, like `pip install 'neuroconv[extract]'`. This is added as a note in the User Guide, as it may be helpful for others. 

## Checklist

- [x] Have you thoroughly read our [Developer Guide](https://neuroconv.readthedocs.io/en/main/developer_guide.html)?
- [x] Have you ensured the PR description clearly describes the problem and solutions?
- [x] Have you checked to ensure that there aren't other open or previously closed [Pull Requests](https://github.com/catalystneuro/neuroconv/pulls) for the same change?
- [x] If this PR fixes an issue, is the first line of the PR description `fix #XXX` where `XXX` is the issue number?
- [x] Did you update the [CHANGLOG.md](https://github.com/catalystneuro/neuroconv/blob/main/CHANGELOG.md) file on your branch to describe your changes?

### CODE REVIEW

File: docs/user_guide/datainterfaces.rst [hunk 1]
```diff
@@ -18,6 +18,15 @@ For instance, to install the dependencies for SpikeGLX, run:
 
     pip install neuroconv[spikeglx]
 
+.. note::
+
+     If you are using `zsh` for command interpreter, you will have to quote the
+     argument like this:
```
Comment (CodyCBakerPhD) [inline docs/user_guide/datainterfaces.rst:24]:
```suggestion
     If you are using a Z-shell (`zsh`) terminal (the default for MacOS), then you will have to use quotes to specify the custom dependency.
```

**Comment by CodyCBakerPhD** [comment]
Small rephrasing otherwise LGTM, thanks for catching and adding this.

**Comment by weiglszonja** [comment]
Thanks @CodyCBakerPhD!

**Review by CodyCBakerPhD — APPROVED** [review]

**Comment by h-mayorquin** [comment]
On the edge of this one, if this is something that we can use in the release notes and users should know it should be there. Things for devops and testing that is mainly for us less clear to me. Documentation probably should go in.

**Pull request merged**